### ISSUE TITLE
How to make pitch shift?

### ISSUE THREAD

**Issue by rafagan** [jamestunnell/spcore#8]
There's some way to make pitch shift using spcore?

**Comment by jamestunnell** [comment]
Do you mean shift pitch of an oscillator? If so, I think you can just modify frequency as you're generate samples. One way would be to use the output of one oscillator to control the frequency input of another oscillator. Here's a pretty basic pitch bend.
```
# Two pitch frequencies
f1 = 440.0 # A4
f2 = 493.88 # B4
srate = 48000
dur_sec = 1.0
nsamples = (srate * dur_sec).to_i
freq_incr = (f2-f1)/nsamples.to_f

osc = Oscillator.new(:sample_rate => srate, :frequency => f1)
output = Array.new(nsamples) do |i|
  x = osc.sample
  osc.frequency += freq_incr
  x
end
```

What's your application, are you generating music?

**Comment by rafagan** [comment]
Actually I’m working on a low latency audio library for android. At this time I have the raw data buffered and the audio thread running, and I would like to use a float value (like 0.9) to adjust audio pitch. But every article I read have a lot of complex content, I also tried a phaser vocoder called rubberband, but it generated a lot of noise in the result.

**Comment by jamestunnell** [comment]
Okay, so would you be running JRuby to be able to use spcore from within Android?

**Comment by rafagan** [comment]
I love ruby and python for prototyping. I’m trying to use spcore to understand the concepts and intuition.

**Comment by jamestunnell** [comment]
That makes sense. Have you read [this article](https://en.wikipedia.org/wiki/Audio_time_stretching_and_pitch_scaling) on time stretching vs. pitch scaling? The spcore library doesn't have any high-level functions like that right now. It can take care of things like interpolation, re-sampling, DFT, etc.

**Comment by rafagan** [comment]
Yes, I've read. But it doesn't demonstrate how to implement pitch shift
